nvi-commons/src/test/java/no/sikt/nva/nvi/common/service/NviServiceTest.java

```diff
@@ -99,10 +100,20 @@ void shouldCreateAndFetchPublicationById() {
 
         var createdCandidate = nviService.upsertCandidate(candidate).orElseThrow();
         var createdCandidateId = createdCandidate.identifier();
+        var fetchedCandidate = nviService.findCandidateById(createdCandidateId).orElseThrow();
 
-        var fetchedCandidate = nviService.findCandidateById(createdCandidateId).orElseThrow().candidate();
+        assertThat(fetchedCandidate.candidate(), is(equalTo(candidate)));
+    }
+
+    @Test
+    void shouldCreateCandidateWithPeriodStatusAndPeriodStatusShouldMatchCorrespondingPeriod() {
+        var year = ZonedDateTime.now().getYear();
+        var period = nviService.getPeriod(String.valueOf(year));
+        var expectedPeriodStatus = new PeriodStatus(period.startDate(), period.reportingDate(), OPEN_PERIOD);
+        var candidate = nviService.upsertCandidate(randomCandidateWithPublicationYear(year)).orElseThrow();
 
-        assertThat(fetchedCandidate, is(equalTo(candidate)));
+        assertThat(candidate.periodStatus().startDate(), is(equalTo(expectedPeriodStatus.startDate())));
```

Comment:
This check is to satisfy jacoco, jacoco says that startDate() method is not in use, while it definitely is.